packages/samples/headless-react/src/pages/DependentFacetPage.tsx

```diff
@@ -0,0 +1,149 @@
+  private readonly fileTypeFacet: HeadlessFacet;
+
+  constructor(props: {}) {
+    super(props);
+
+    this.engine = buildSearchEngine({
+      configuration: getSampleSearchEngineConfiguration(),
+    });
+
+    this.authorFacet = buildFacet(this.engine, {options: {field: 'author'}});
+    this.sourceFacet = buildFacet(this.engine, {options: {field: 'source'}});
+    this.objectTypeFacet = buildFacet(this.engine, {
+      options: {field: 'objecttype'},
+    });
+    this.fileTypeFacet = buildFacet(this.engine, {
+      options: {field: 'filetype'},
+    });
+    this.state = {currentExample: 'SingleParentSingleDependent'};
+  }
+
+  componentDidMount() {
+    this.engine.executeFirstSearch();
+  }
+
+  render() {
+    return (
+      <>
+        <select
+          onChange={(e) =>
+            this.setState({currentExample: e.target.value as ExampleType})
+          }
+        >
+          {Object.entries(Examples).map(([key, value]) => (
+            <option value={key} key={key}>
+              {value.name}
+            </option>
+          ))}
+        </select>
+        <div>
+          <h2>{Examples[this.state.currentExample].name}</h2>
+          <h3>{Examples[this.state.currentExample].description}</h3>
+          {this.renderExample()}
+        </div>
+      </>
+    );
+  }
+
+  renderExample() {
+    switch (this.state.currentExample) {
+      case 'SingleParentSingleDependent':
+        return (
+          <SingleParentSingleDependentFacet
+            parentFacet={this.sourceFacet}
+            dependentFacet={this.objectTypeFacet}
+          />
+        );
+      case 'SingleParentMultipleDependent':
+        return (
+          <SingleParentMultipleDependentFacet
+            parentFacet={this.sourceFacet}
+            dependentFacets={[this.objectTypeFacet, this.fileTypeFacet]}
+          />
+        );
+
+      case 'SingleValueDependent':
+        return (
+          <SingleValueDependentFacet
+            parentFacet={this.fileTypeFacet}
+            dependentFacet={this.authorFacet}
+            dependentValue="doc"
+          />
+        );
+
+      case 'ComplexDependencies':
+        return (
+          <ComplexDependentFacet
+            dependencies={{
+              [this.sourceFacet.state.facetId]: {
+                facet: this.sourceFacet,
+                controls: [this.objectTypeFacet, this.fileTypeFacet],
+              },
+              [this.objectTypeFacet.state.facetId]: {
+                facet: this.objectTypeFacet,
+                dependsOn: this.sourceFacet,
+              },
+              [this.authorFacet.state.facetId]: {
+                facet: this.authorFacet,
+                dependsOn: this.sourceFacet,
+                controls: [this.authorFacet],
+              },
+              [this.fileTypeFacet.state.facetId]: {
+                facet: this.fileTypeFacet,
+                dependsOn: this.authorFacet,
+              },
+            }}
+          />
+        );
```

Comment:
I wonder if there is a design where the relationships can be shown declaratively:

```
<DependentFacetManager>
  <Facet id='a'>
     <Facet id='b'/>
     <CategoryFacet id='c'/>
  </Facet>
</DependentFacetManager>
```

I'm not sure how/if such api could be implemented. If it is possible, it would help visualize the relationships, and provide flexibility around the kind of facets being linked (atm, it's only `Facet`).

Might be worth investigating if we one day provide a `@coveo/react-ui` package, but I think for educational purposes, the current samples are clearer.